source/h1_stream.c

```diff
@@ -74,16 +79,61 @@ static void s_stream_cross_thread_work_task(struct aws_channel_task *task, void
     }
 
     if (new_outgoing_data && (api_state == AWS_H1_STREAM_API_STATE_ACTIVE)) {
-        aws_h1_connection_try_write_outgoing_stream(s_get_h1_connection(stream));
+        aws_h1_connection_try_write_outgoing_stream(connection);
+    }
+
+    /* Add to window size using saturated sum to prevent overflow.
+     * Saturating is fine because it's a u64, the stream could never receive that much data. */
+    stream->thread_data.window_size = aws_add_u64_saturating(stream->thread_data.window_size, pending_window_update);
+    if ((pending_window_update > 0) && (api_state == AWS_H1_STREAM_API_STATE_ACTIVE)) {
+        /* Now that stream window is larger, connection might have buffered
+         * data to send, or might need to increment its own window */
+        aws_h1_connection_try_process_read_messages(connection);
```

Comment:
It might be better to early stop here if the stream is not the current incoming stream?

Reply:
eh, that's complicated. the function can handle it
doesn't seem worth optimizing for since it's pretty rare for a connection to have multiple streams